src/ClientData/CaptureDataTest.cpp

```diff
@@ -0,0 +1,96 @@
+
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <iterator>
+#include <optional>
+
+#include "ClientData/CaptureData.h"
+#include "ClientData/ScopeIdConstants.h"
+#include "ClientData/ScopeIdProvider.h"
+#include "ClientData/ScopeStats.h"
+#include "GrpcProtos/capture.pb.h"
+#include "SharedTestConstants.h"
+
+namespace orbit_client_data {
+
+class MockScopeIdProvider : public ScopeIdProvider {
```

Comment:
Nit: I think technically, your usage here is not a mock but a fake and you could simply add a fake implementation of ScopeIdProvider (without the MOCK_METHOD). Feel free to ignore.

Reply:
But I'll also have to fake/mock `GetScopeName`. OTOH, it will be a one-liner returning an empty string.